server/src/kernels.cu

```diff
@@ -345,24 +518,88 @@ void diff::cuda::CUDACore::exec_core(uint8_t *frameData, uint8_t *showReadyNData
         kernel2_char<<<1, nThreadsToUse>>>(d_current, d_charsPx + idx * fullArea, eachThreadDoes, offset, 3 * charsSz.width, 3 * frameSz.width);
     }
 
+#ifdef NOISE_VISUALIZER
+#if NOISE_VISUALIZER == 4
+
+    // grayscale_kernel<<<1, nMaxThreads>>>(d_current, d_grayscale, maxAtTime);
+    // grayscale_kernel_v2<<<1, nMaxThreads>>>(d_current, d_grayscale, max4);
+    grayscale_kernel_v3<<<1, nMaxThreads>>>(d_current, d_grayscale, max4);
+    cudaMemcpyAsync(showReadyNData, d_grayscale, total, cudaMemcpyDeviceToHost);
+
+#elif NOISE_VISUALIZER == 5
+    int h_histogram[256] = {0};
+    uint8_t h_indexes[2] = {0};
+    int index_max = -1, index_sec_max = -1;
+
+    cudaMemcpyAsync(d_histogram, h_histogram, 256 * sizeof(int), cudaMemcpyHostToDevice);
+    grayscale_kernel_v3<<<1, nMaxThreads>>>(d_current, d_grayscale, max4);
+    // generate_histogram<<<1, nMaxThreads>>>(d_grayscale, d_histogram, maxAtTime);
+    generate_histogram_v2<<<1, nMaxThreads>>>(d_grayscale, d_histogram, maxAtTime);
+    compute_max<<<1, 256>>>(d_histogram, d_indexes_max);
+    // cudaMemcpyAsync(showReadyNData, d_grayscale, total, cudaMemcpyDeviceToHost);
+    cudaMemcpyAsync(h_histogram, d_histogram, 256 * sizeof(int), cudaMemcpyDeviceToHost);
+    cudaMemcpyAsync(h_indexes, d_indexes_max, 2 * sizeof(uint8_t), cudaMemcpyDeviceToHost);
+
+    // printf("\nIstogramma\n");
+
+    // for (int i = 0; i < 256; i++) {
+    //     printf("%d\n", h_histogram[i]);
+    // }
+    // printf("\nMassimi\n");
+
+    index_max = h_indexes[0];
+    index_sec_max = h_indexes[1];
+    // // printf("%d %d\n", index_max, index_sec_max);
```

Comment:
If the `printf` are useless, you can remove them to have a cleaner code

Reply:
Yes, good suggestion. Now they are removed